### ISSUE TITLE
Emulation breaks when running CEmu in a VM

### ISSUE THREAD

**Issue by alberthdev** [CE-Programming/CEmu#62]
When running CEmu in a VM, emulation becomes wonky... with the latest VirtualBox and Windows 7, the emulation is ridiculously fast (and unusable), and the speed throttling has no effect. (I haven't tested this with any other OSes like Linux/Mac, or even other VM software, so YMMV.)

This is due to using `std::chrono::steady_clock`, which the emulator uses for high precision timing. This particular library (or maybe even compiler MSVC2015) directly uses the CPU's high precision timers as a source. VirtualBox (and maybe other emulators?) have some trouble with making accurate measurements of time, so this results in poor emulation behavior on the VM.

As a workaround, I've moved from `std::chrono::steady_clock` to `QElapsedTimer`. It seems that the Qt timers work fine on a VM (maybe they have workarounds?), and emulation works properly on my VM with Qt's timers - at regular speed, with the throttle slider working.

As an aside, it may also be possible to enable HPET on VirtualBox (and other VMs) to mitigate this issue. However, it may not be very straightforward - VirtualBox doesn't even expose this option with their normal settings dialog. (You will have to use `VBoxManage modifyvm VM_NAME --hpet on` to enable it on VirtualBox.)

Since this is somewhat of a big change, I've posted this change in a separate branch: [`vm-badtimer-support`](https://github.com/CE-Programming/CEmu/tree/vm-badtimer-support).

**Comment by MateoConLechuga** [comment]
Merged [`vm-badtimer-support`](https://github.com/CE-Programming/CEmu/tree/vm-badtimer-support) with master branch. Thanks for your help! The RTC is now confirmed to work properly as well :)

**Comment by alberthdev** [comment]
As an interesting bump to this bug... so the [`vm-badtimer-support`](https://github.com/CE-Programming/CEmu/tree/vm-badtimer-support) got reverted a while ago due to some issues with the patch. That said, as of right now, using the latest [67384fd](https://oss.jfrog.org/artifactory/oss-snapshot-local/org/github/alberthdev/cemu/git/201610/20161024/cemu-20161024_001349-git-67384fd/) build, it looks like the emulation is working fine at the correct speed. Not sure if this was a code change that caused it, or maybe the fact that we are no longer using the QtQuick keypad...

Either way, for those who are wondering... this bug has a happy ending!

**Comment by Schnurrwuschel** [comment]
With VMware+Ubuntu 18.04 and 20.4 Cemu is not working with Arm-Thumb. Arm64 is working in Step-mode, but freezing in run-mode. Arm and Arm-Thumb is not working in either modes. 
 I found no way to use your changes. The  vm-badtimer-support is not available. I installed the virtualbox in W7 and Ubuntu. Did you use - VBoxManage modifyvm VM_NAME --hpet on - inside the Virtualbox in W7 or inside VM-Ubuntu? I had no reaction and no error. Using the latest 67384fd build is not using VM. It is an *.exe-file for using inside Windows.
Which dialog should I find in the Virtualbox?
Github:
As a workaround, I've moved from std::chrono::steady_clock to QElapsedTimer. It seems that the Qt timers work fine on a VM (maybe they have workarounds?), and emulation works properly on my VM with Qt's timers - at regular speed, with the throttle slider working.

As an aside, it may also be possible to enable HPET on VirtualBox (and other VMs) to mitigate this issue. However, it may not be very straightforward - VirtualBox doesn't even expose this option with their normal settings dialog. (You will have to use VBoxManage modifyvm VM_NAME --hpet on to enable it on VirtualBox.)
